client/src/visualiser-src/common/GraphicalDataStructureFactory.ts

```diff
@@ -17,6 +17,25 @@ class GraphicalDataStructureFactory {
         throw Error('Invalid Topic Title');
     }
   }
+
+  public static generate(topicTitle: string): GraphicalDataStructure {
```

Comment:
I don't think it's necessary to reinstantiate another data structure object -- we can simply call reset and then another 'randomlyGenerate' method.